doc/ifa/SONiC_InbandFlowAnalyzer_FeatureSpec_Ver1.1.md

```diff
@@ -0,0 +1,478 @@
+
+### 1.2.2 Container
+No new containers are added for Inband Flow Analyzer (IFA) feature.
+
+### 1.2.3 SAI Overview
+
+The SAI TAM spec specifies the TAM APIs to be used to configure the IFA functionality. Please refer to SAI-Proposal-TAM2.0-v2.0.docx in [https://github.com/opencomputeproject/SAI/tree/master/doc/TAM](https://github.com/opencomputeproject/SAI/tree/master/doc/TAM) for more details.
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+
+IFA is used to proactively monitor the network for faults and performance bottlenecks, it aids in detecting data plane faults and isolating them to a given device or location and assists in planning network capacity and projecting future load.
+
+## 2.2 Functional Description
+
+The IFA feature allows user to configure a device as ingress, intermediate and egress nodes through PACKET\_ACTION field in ACL rule. PACKET\_ACTION field values INT\_INSERT, INT\_UPDATE and INT\_DELETE specifies ingress, intermediate and egress devices respectively. Each device type performs its own unique operations as mentioned below.
+
+Ingress node matches an incoming flow based on the ACL configuration specified in an IFA flow and makes a copy of a sampled packet from the flow based upon the configured sampling rate. The copied packet has the IFA meta-data instructions and IFA header and IFA meta-header data added, and is then forwarded along the same path as the sampled packet. 
+
+At intermediate nodes, IFA probe packets are identified and IFA meta-data is inserted and forwarded to next hop. IFA meta-data consists of fields such as IFA probe packet's ingress port, egress port, ingress time-stamp and egress time-stamp etc.
+
+At egress node, IFA probe packets are identified and IFA meta-data is inserted. Egress node terminates IFA session and summarizes the telemetry data for the entire path, encapsulates the telemetry data as payload and sends it to collector.
+
+At all other non-IFA nodes, IFA sampled packet is forwarded as a normal packet in the network.
+
+***Example IFA multinode topology***
+
+![IFA multinode topology](images/IFA_Multinode_Topology.JPG)
+
+***Example CLI configuration***
+
+    ***Ingress node***
+
+        IFA configuration
+            config ifa device-id 2345
+            config ifa flow-name flow1 acl_ifa 1000
+
+        ACL configuration
+            "ACL_TABLE": {
+                "IFA": {
+                    "policy_desc" : "IFA Ingress Device Policy",
+                    "stage" : "INGRESS",
+                    "type" : "L3" ,
+                    "ports" : "Ethernet20"
+                    }
+            }
+
+            "ACL_RULE": {
+                "IFA|Flow1": {
+                    "PRIORITY" : "55",
+                    "IP_TYPE" : "ipv4any",
+                    "SRC_IP" : "10.10.0.26/32",
+                    "DST_IP" : "10.10.1.26/32",
+                    "PACKET_ACTION" : "int_insert"
```

Comment:
Why not use the command ``config ifa device-type ingress`` to configure a device as ingress?